### PR TITLE
feat: bbox shrinking in xycut algo, better natural reading order

### LINKED ISSUE #1233: feat: more robust xycut's by pre/post processing bboxes
**Is your feature request related to a problem? Please describe.**
The xycut algorithm works well when vertical or horizontal lines can be drawn through whitespace as part of its divide-and-conquer recursive solution. However, if the bounding boxes the algo is given overlap or are flush against each other, it is not going to do well (because it can't draw lines with no whitespace between the bboxes).

E.g.
image
from https://colab.research.google.com/drive/1q-wjJdFjGhhei5C9m-hzXszYANrXuWbT.

**Describe the solution you'd like**
One quick workaround that shouldn't negatively affect cases where the algo is working well would be to shrink all bounding boxes by 90% along x and y axes (still centered around the same center point). So, shrink bboxes before algo, run sort, revert bboxes to original sizes.

This is lower priority than https://github.com/Unstructured-IO/unstructured/issues/1209 .

### PR DESCRIPTION
**Pull request by christinestraub**
Closes #1233.
### Summary
- add functionality to shrink all bounding boxes along x and y axes (still centered around the same center point) before running xy-cut sort
### Evaluation
Run the followin gcommand for this [PDF](https://utic-dev-tech-fixtures.s3.us-east-2.amazonaws.com/pastebin/patent-11723901-page2.pdf).
```
PYTHONPATH=. python examples/custom-layout-order/evaluate_xy_cut_sorting.py <file_path> <strategy>
```

### CODE REVIEW

File: unstructured/partition/utils/constants.py [hunk 1]
```diff
@@ -1,2 +1,3 @@
 SORT_MODE_XY_CUT = "xy-cut"
 SORT_MODE_BASIC = "basic"
+XY_CUT_BBOX_SHRINK_FACTOR = 0.9
```
Comment (cragwolfe) [inline unstructured/partition/utils/constants.py:3]:
```suggestion
XY_CUT_BBOX_SHRINK_FACTOR = 0.9
```
please make this the default but make configurable via env var

**Comment by cragwolfe** [comment]
@christinestraub , please update the ingest-test fixtures (using the "Run workflow" button for your branch here 
https://github.com/Unstructured-IO/unstructured/actions/workflows/ingest-test-fixtures-update-pr.yml )

**Comment by christinestraub** [comment]
> @christinestraub , please update the ingest-test fixtures (using the "Run workflow" button for your branch here https://github.com/Unstructured-IO/unstructured/actions/workflows/ingest-test-fixtures-update-pr.yml )

Done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ### Enhancements
 
+* **Improve `xy-cut` sorting by pre/post processing bboxes**. The new behavior is to shrink all bounding boxes by 90% along x and y axes (still centered around the same center point).
```
Comment (cragwolfe) [inline CHANGELOG.md:5]:
```suggestion
* **Better detection of natural reading order in images and PDF's** Improves `xy-cut` sorting by pre/post processing bboxes. The new behavior is to shrink all bounding boxes by 90% along x and y axes (still centered around the same center point), which allows projection lines to be drawn where not possible before if layout bboxes overlapped.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ### Enhancements
 
+* **Better detection of natural reading order in images and PDF's** Improves `xy-cut` sorting by pre/post processing bboxes. The new behavior is to shrink all bounding boxes by 90% along x and y axes (still centered around the same center point), which allows projection lines to be drawn where not possible before if layout bboxes overlapped.
```
Comment (cragwolfe) [inline CHANGELOG.md:5]:
```suggestion
* **Better detection of natural reading order in images and PDF's** The elements returned by partition better reflect natural reading order in some cases, particularly in complicated multi-column layouts, leading to better chunking and retrieval for downstream applications. Achieved by improving the `xy-cut` sorting to preprocess bboxes, shrinking all bounding boxes by 90% along x and y axes (still centered around the same center point), which allows projection lines to be drawn where not possible before if layout bboxes overlapped.
```

**Review by cragwolfe — APPROVED** [review]
The 4 changes in the IRS doc test fixture update were all improvements. The layout-parser paper results were mixed, presumably because indentation changes for bboxes of differing widths. Overall an improvement! LGTM

**Pull request merged**